pop_tools/fill.py

```diff
@@ -183,3 +223,93 @@ def _iterative_fill_POP_core(nlat, nlon, var, fillmask, missing_value, tol, ltri
                             done = False
 
         var[1:nlat, :] = work[1:nlat, :]
+
+
+@jit(nopython=True)
+def _iterative_fill_sor(
+    nlat, nlon, var, fillmask, tol=5.0e-4, rc=1.6, max_iter=100, ltripole=False
+):
+    """Iterative land fill algorithm via SOR solution of Laplace Equation."""
+
+    # Compute a zonal mean to use as a first guess
+    # Apprarently jit doesn't like masked arrays so loop it out
+    zoncnt = np.zeros(nlat)
+    zonavg = np.zeros(nlat)
+    for j in range(0, nlat):
+        zoncnt[j] = np.sum(np.where(fillmask[j, :], 0, 1))
+        zonavg[j] = np.sum(np.where(fillmask[j, :], 0, var[j, :]))
+        if zoncnt[j] != 0:
+            zonavg[j] = zonavg[j] / zoncnt[j]
+
+    # Fill missing zonal averages for rows that are entirely land
+    for j in range(0, nlat - 1):  # northward pass
+        if zoncnt[j] > 0 and zoncnt[j + 1] == 0:
+            zoncnt[j + 1] = 1
+            zonavg[j + 1] = zonavg[j]
+    for j in range(0, nlat - 1):  # southward pass
+        jrev = nlat - 1 - j
+        if zoncnt[jrev] > 0 and zoncnt[jrev - 1] == 0:
+            zoncnt[jrev - 1] = 1
+            zonavg[jrev - 1] = zonavg[jrev]
```

Comment:
I am curious... Isn't it the case that these two loops can be merged into a single loop?

Reply:
I don't think so -- you want the southward pass to act on the values modified by the northward pass. Say the northward pass modifies `zonavg[j]` for a `j` close to `nlat`; the southward pass uses that value for `j` close to 0 (=> `jrev` close to `nlat`), and if it's all one loop the southward pass will use the unmodified value.

It might be clearer if the second loop was `for jrev in range(nlat-2, -1, -1):` (or drop the `jrev` notation and just use `j` everywhere)

Reply:
> It might be clearer if the second loop was for jrev in range(nlat-2, -1, -1): (or drop the jrev notation and just use j everywhere)

Make sense.... `for j in range(0, nlat - 1):` is what led me to believe that the two loops could be merged.